core/server/master/src/main/java/alluxio/master/MasterProcess.java

```diff
@@ -92,4 +95,29 @@ private Factory() {} // prevent instantiation
    * @return configuration information list
    */
   List<ConfigProperty> getConfiguration();
+
+  /**
+   * Returns a master id for the given master, creating one if the master is new.
+   *
+   * @param hostname the master hostname
+   * @return the master id for this master
+   */
+  long getMasterId(String hostname);
```

Comment:
can you add a TODO somewhere in this file that we should have a separate `MetaMaster` interface and implementation, rather than free-riding `MasterProcess`

Reply:
Added, thanks!